### ISSUE TITLE
Using unordered_map and unordered_set together may cause an error

### ISSUE THREAD

**Issue by tenk-a** [LADSoft/OrangeC#45]
Using unordered_map and unordered_set together may cause an error

smp.c:
```
#include <unordered_map>
#include <unordered_set>
#include <utility>
int main() {
    std::unordered_map<int,int> m1;
    std::unordered_map<int,int> m2;
    std::swap(m1,m2);
    std::unordered_set<int>     s1;
    std::unordered_set<int>     s2 = s1;
}
```
RESULT:
Error(215)    C:\Program Files (x86)\Orange C 386\include\memory(2325):  Ambiguity between 'std::__1::forward<_Tp>(remove_reference<_Tp>::type&)' and 'std::__1::forward<_Tp>(remove_reference<_Tp>::type&&)' in function std::__1::__compressed_pair<unsigned int, hash<int>>::__compressed_pair<unsigned int, hash<int>>(unsigned int, hash<int>)
Error(215)    C:\Program Files (x86)\Orange C 386\include\memory(1963):  Ambiguity between 'std::__1::forward<_Tp>(remove_reference<_Tp>::type&)' and 'std::__1::forward<_Tp>(remove_reference<_Tp>::type&&)' in function std::__1::__libcpp_compressed_pair_imp<unsigned int, hash<int>, int=0>::__libcpp_compressed_pair_imp<unsigned int, hash<int>, int=0>(unsigned int, hash<int>)
Error(240)    C:\Program Files (x86)\Orange C 386\include\__hash_table(2131):  Reference to deleted function 'std::__1::unique_ptr<__hash_node<int, void*>, __hash_node_destructor<allocator<__hash_node<int, void*>>>>::unique_ptr<__hash_node<int, void*>, __hash_node_destructor<allocator<__hash_node<int, void*>>>>( const unique_ptr<__hash_node<int, void*>, __hash_node_destructor<allocator<__hash_node<int, void*>>>>&)' in function std::__1::__hash_table<int, hash<int>, equal_to<int>, allocator<int>>::__construct_node( const int&, unsigned int)
Warning(436)  smp.cpp(9):  In template instantiation started here

**Comment by LADSoft** [comment]
fixed in 6.0.28

**Comment by tenk-a** [comment]
I compiled with v 6.0.28.1.
Thank you.
